### ISSUE TITLE
🐛 [cloud_firestore] [iOS] [Emulator] Firestore Emulator gives: permission-denied, The caller does not have permission to execute the specified operation

### ISSUE THREAD

**Issue by brandonhub** [FirebaseExtended/flutterfire#5468]
## Bug report
Hi everyone!

When I try to connect my iOS **simulator** to the firebase **emulator** I get  the following error when I query any document:
```
PlatformException (PlatformException(permission-denied, The caller does not have permission to execute the specified operation., {code: permission-denied, message: The caller does not have permission to execute the specified operation., nativeErrorMessage: Missing or insufficient permissions., nativeErrorCode: 7}, null))
```

My rules are completely open:
```
rules_version = '2';

// Allow read/write access to all users under any conditions
// Warning: **NEVER** use this rule set in production; it allows
// anyone to overwrite your entire database.
service cloud.firestore {
  match /databases/{database}/documents {
    match /{document=**} {
      allow read, write: if true;
    }
  }
}
```

Here are my relevant pubspec.yml file...
```
version: 1.0.0+1

environment:
  sdk: ">=2.2.2 <3.0.0"

dependencies:
  # Core
  flutter:
    sdk: flutter

  firebase_messaging: ^9.0.0
  firebase_analytics: ^7.1.1
  google_sign_in: ^5.0.0
  firebase_auth: ^1.0.1
  firebase_core: ^1.0.1
  cloud_firestore: ^1.0.1
  firebase_remote_config: ^0.9.0-dev.0
  firebase_crashlytics: ^1.0.0

```

Here is how I connect the flutter app to emulator (I call this after I call `Firebase.initializeApp();`):
```
FirebaseFirestore _getFirestoreInstance() {
  final firestoreInstance = FirebaseFirestore.instance;
  firestoreInstance.settings = Settings(
    persistenceEnabled: false,
    sslEnabled: false,
    host: '127.0.0.1:8080',
  );
  return firestoreInstance;
}
```

I have tried replacing host with `0.0.0.0` and `localhost`, but same issue either way.

### Steps to reproduce

1. Start firebase emulator
1. Run iOS simulator and connect it with firebase emulator (see code snippet above)
2. Query any document in firestore (my rules are completely open)
4. See the error

### Expected behavior

My rules are open so I should be able to query anything.

---

## Additional context

The error code 7 is interesting. Wish I knew what `nativeErrorCode: 7` meant...

Also, I have the following line in my Podfile in order to speed up build times. But still same issue if I remove this line:
```
pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git', :tag => '7.3.0'
```

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 2.0.2, on Mac OS X 10.15.7 19H524 darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.4)
[✓] VS Code (version 1.54.3)
[✓] Connected device (2 available)
    ! Error: Brandon's iPhone 8 is not connected. Xcode will continue when Brandon's iPhone 8 is connected. (code -13)
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.12.1
Flutter SDK 2.0.2

dependencies:
- app_settings 4.1.0 [flutter]
- cloud_firestore 1.0.1 [cloud_firestore_platform_interface cloud_firestore_web firebase_core firebase_core_platform_interface flutter meta]
- cupertino_icons 1.0.2
- firebase_analytics 7.1.1 [firebase_analytics_platform_interface firebase_analytics_web firebase_core flutter meta]
- firebase_auth 1.0.1 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 1.0.1 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 1.0.0 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- firebase_messaging 9.0.0 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_remote_config 0.9.0-dev.0 [flutter firebase_core firebase_core_platform_interface firebase_remote_config_platform_interface]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_background_geolocation 1.7.0 [flutter background_fetch]
- flutter_spinkit 5.0.0 [flutter]
- flutter_webview_plugin 0.3.11 [flutter]
- google_maps_flutter 2.0.1 [flutter flutter_plugin_android_lifecycle google_maps_flutter_platform_interface]
- google_sign_in 5.0.0 [google_sign_in_platform_interface google_sign_in_web flutter meta]
- http 0.13.0 [http_parser meta path pedantic]
- intl 0.17.0 [clock path]
- location 4.1.1 [flutter location_platform_interface location_web meta]
- location_permissions 3.0.0+1 [flutter meta]
- overlay_support 1.2.0 [flutter async]
- package_info 2.0.0 [flutter]
- permission_handler 6.1.0 [flutter meta permission_handler_platform_interface]
- provider 5.0.0 [collection flutter nested]
- qr_code_scanner 0.3.5 [flutter]
- qr_flutter 4.0.0 [flutter qr]
- rxdart 0.26.0
- share 2.0.1 [meta mime flutter]
- shared_preferences 2.0.4 [meta flutter shared_preferences_platform_interface shared_preferences_linux shared_preferences_macos shared_preferences_web shared_preferences_windows]
- url_launcher 6.0.2 [flutter url_launcher_platform_interface url_launcher_linux url_launcher_macos url_launcher_windows url_launcher_web]
- uuid 3.0.1 [crypto]
- video_player 2.0.2 [meta video_player_platform_interface video_player_web flutter flutter_test]
- wakelock 0.5.0+2 [flutter meta wakelock_macos wakelock_windows wakelock_platform_interface wakelock_web]

dev dependencies:
- flutter_launcher_icons 0.9.0 [args image path yaml]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- archive 3.1.2 [crypto path]
- args 2.0.0
- async 2.5.0 [collection]
- background_fetch 0.5.6 [flutter]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.1.0
- charcode 1.2.0
- clock 1.1.0
- cloud_firestore_platform_interface 4.0.0 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 1.0.1 [cloud_firestore_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.15.0
- crypto 3.0.0 [collection typed_data]
- fake_async 1.2.0 [clock collection]
- ffi 1.0.0
- file 6.1.0 [meta path]
- firebase 9.0.0 [http http_parser js]
- firebase_analytics_platform_interface 1.1.0 [flutter meta]
- firebase_analytics_web 0.2.0+1 [firebase firebase_analytics_platform_interface flutter flutter_web_plugins meta]
- firebase_auth_platform_interface 4.0.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 1.0.2 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.0.0 [flutter meta plugin_platform_interface]
- firebase_core_web 1.0.1 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_crashlytics_platform_interface 2.0.0 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_platform_interface 2.0.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 1.0.1 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- firebase_remote_config_platform_interface 0.2.0-dev.0 [flutter meta firebase_core plugin_platform_interface]
- flutter_plugin_android_lifecycle 2.0.0 [flutter]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- google_maps_flutter_platform_interface 2.0.1 [flutter meta plugin_platform_interface stream_transform collection]
- google_sign_in_platform_interface 2.0.1 [flutter meta quiver]
- google_sign_in_web 0.10.0 [google_sign_in_platform_interface flutter flutter_web_plugins meta js]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image 3.0.1 [archive meta xml]
- js 0.6.3
- location_platform_interface 2.1.0 [flutter meta plugin_platform_interface]
- location_web 3.0.0 [flutter flutter_web_plugins http_parser js location_platform_interface meta]
- matcher 0.12.10 [stack_trace]
- meta 1.3.0
- mime 1.0.0
- nested 1.0.0 [flutter]
- path 1.8.0
- path_provider_linux 2.0.0 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_platform_interface 2.0.1 [flutter meta platform plugin_platform_interface]
- path_provider_windows 2.0.0 [path_provider_platform_interface meta path flutter ffi win32]
- pedantic 1.11.0
- permission_handler_platform_interface 3.1.0 [flutter meta plugin_platform_interface]
- petitparser 4.0.2 [meta]
- platform 3.0.0
- plugin_platform_interface 2.0.0 [meta]
- process 4.1.0 [file path platform]
- qr 2.0.0 [meta]
- quiver 3.0.0 [matcher]
- shared_preferences_linux 2.0.0 [flutter file meta path path_provider_linux shared_preferences_platform_interface]
- shared_preferences_macos 2.0.0 [shared_preferences_platform_interface flutter]
- shared_preferences_platform_interface 2.0.0 [flutter]
- shared_preferences_web 2.0.0 [shared_preferences_platform_interface flutter flutter_web_plugins meta]
- shared_preferences_windows 2.0.0 [shared_preferences_platform_interface flutter file meta path path_provider_platform_interface path_provider_windows]
- sky_engine 0.0.99
- source_span 1.8.0 [charcode collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- stream_transform 2.0.0
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.2.19 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0 [collection]
- url_launcher_linux 2.0.0 [flutter]
- url_launcher_macos 2.0.0 [flutter]
- url_launcher_platform_interface 2.0.2 [flutter plugin_platform_interface]
- url_launcher_web 2.0.0 [url_launcher_platform_interface meta flutter flutter_web_plugins]
- url_launcher_windows 2.0.0 [flutter]
- vector_math 2.1.0
- video_player_platform_interface 4.0.0 [flutter meta flutter_test]
- video_player_web 2.0.0 [flutter flutter_web_plugins meta video_player_platform_interface]
- wakelock_macos 0.1.0 [flutter flutter_web_plugins wakelock_platform_interface]
- wakelock_platform_interface 0.2.0 [flutter meta]
- wakelock_web 0.2.0 [flutter flutter_web_plugins js wakelock_platform_interface]
- wakelock_windows 0.1.0 [flutter wakelock_platform_interface win32]
- win32 2.0.4 [ffi]
- xdg_directories 0.2.0 [meta path process]
- xml 5.0.2 [collection meta petitparser]
- yaml 3.1.0 [collection source_span string_scanner]
```

</details>

---

**Comment by markusaksli-nc** [comment]
Hi @brandonhub 
Have you done https://firebase.flutter.dev/docs/installation/ios#enabling-use-of-firebase-emulator-suite
I'm not seeing this issue.
Thank you

**Comment by MeAtPros** [comment]
Any update ???

**Comment by brandonhub** [comment]
Hi @markusaksli-nc 

Thanks for the quick response! Yeah, I have that entry in my Info.plist and still no luck :/ 

Any idea what the error code 7 represents I firebase?

**Comment by markusaksli-nc** [comment]
Not sure, labeling this for some further investigation since I wasn't able to reproduce it

**Comment by brandonhub** [comment]
Alright, I've made a dumb mistake. The issue was that I was correctly setting the Firestore settings like so...
```
FirebaseFirestore.instance.settings = Settings(host: host, sslEnabled: false);
```
but then later in my code I was setting the settings *again* like this...
```
FirebaseFirestore.instance.settings = Settings(persistenceEnabled: false);
```
thus overriding my original settings.

This caused flutter app to connect to the remote Firestore instance instead of the emulator, but attempt to run queries using emulated auth token, which obviously failed.

Closing, sorry for the noise.
